File: Minimum-Absolute-Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        vector<vector<int>> pairs={{arr[0],arr[1]}};
        int minabs=arr[1]-arr[0];
        for(int i=2;i<arr.size();i++){
            if(arr[i] - arr[i-1] < minabs ){
                pairs.clear();
                minabs=arr[i] - arr[i-1];
                pairs.push_back({arr[i-1],arr[i]});
            }
            else if (arr[i] - arr[i-1] ==minabs)pairs.push_back({arr[i-1],arr[i]});
        }
        return pairs;
    }
};

File: Minimum-Absolute-Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr); // Sort the array
        List<List<Integer>> pairs = new ArrayList<>();
        int minAbs = Integer.MAX_VALUE;

        // Find the minimum absolute difference
        for (int i = 1; i < arr.length; i++) {
            int diff = arr[i] - arr[i - 1];
            if (diff < minAbs) {
                pairs.clear(); // Clear the list if a new minimum is found
                minAbs = diff; // Update the minimum difference
                pairs.add(Arrays.asList(arr[i - 1], arr[i])); // Add the new pair
            } else if (diff == minAbs) {
                pairs.add(Arrays.asList(arr[i - 1], arr[i])); // Add the pair if it matches the minimum
            }
        }

        return pairs; // Return the list of pairs
    }
}